Controller/Advice/Call.php

```diff
@@ -0,0 +1,126 @@
+                'auth_type' => 'fraud',
+                'status' => $status
+            );
+            $logMessage = sprintf(__('advise_log_quote_denied'), $quoteId);
+            //saves advise call returned data in quote Payment (additional data)
+            $this->updateQuotePaymentDetailsInDb($quoteId, $paymentDetails);
+            //Riskified defined order as fraud - order data is send to Riskified
+            $this->sendDeniedOrderToRiskified($quote);
+            $this->logger->log($logMessage);
+            $message = __('checkout_declined');
+            $this->messageManager->addError(__('checkout_declined'));
+        }else {
+            $paymentDetails = array(
+                'date' => $currentDate = date('Y-m-d H:i:s', time()),
+                'auth_type' => $authType,
+                'status' => $status
+            );
+            if($authType == "sca"){
+                $adviceCallStatus = false;
+                $message = __('advice_sca');
+            }else{
+                $adviceCallStatus = true;
+                $message = __('advice_tra');
+            }
+            //saves advise call returned data in quote Payment (additional data)
+            $this->updateQuotePaymentDetailsInDb($quoteId, $paymentDetails);
+        }
+
+        return  $this->resultJsonFactory->create()->setData(['advice_status' => $adviceCallStatus, 'message' => $message]);
+    }
+
+    /**
+     * Saves quote payment details (additional data).
+     * @param $quoteId
+     * @param $paymentDetails
+     * @throws \Exception
+     */
+    protected function updateQuotePaymentDetailsInDb($quoteId, $paymentDetails)
+    {
+        $quote = $this->registry->registry($quoteId);
+        if (isset($quote)) {
+            $this->logger->log(sprintf(__('advise_log_quote_save'), $quoteId));
+            $quotePayment = $quote->getPayment();
+            $additionalData = $quotePayment->getAdditionalData();
+
+            //avoid overwriting quotePayment additional data
+            if (!is_array($additionalData)) {
+                $additionalData = [];
+            }
+            $additionalData[$paymentDetails['auth_type']] = $paymentDetails;
+            $additionalData = json_encode($additionalData);
+            try {
+                $quotePayment->setAdditionalData($additionalData);
+                $quotePayment->save();
+            } catch (RuntimeException $e) {
+                $this->logger->log(sprintf(__('advise_log_no_quote_found'), $e->getMessage()));
+            }
+        } else {
+            $this->logger->log(sprintf(__('advise_log_no_quote_found'), $quoteId));
+        }
+    }
+
+    /**
+     * @param $quote
+     * @return mixed
+     */
+    protected function sendDeniedOrderToRiskified($quote)
+    {
+        return parent::sendDeniedOrderToRiskified($quote);
+    }
+
+    /**
+     * @param $cartId
+     * @return mixed
+     */
+    protected function getQuoteId($cartId)
```

Comment:
Why do you need this method?

Reply:
function for any case to find quote id. In case order is made by anonymous guest quote no is masked.